core/services/functions/listener.go

```diff
@@ -386,32 +398,29 @@ func (l *FunctionsListener) getMaxSecretsSize(flags RequestFlags) uint32 {
 	return l.pluginConfig.MaxSecretsSizesList[idx]
 }
 
-// TODO (FUN-662): call from LogPoller in a separate goroutine
-func (l *FunctionsListener) HandleOracleRequestV1(request *functions_coordinator.FunctionsCoordinatorOracleRequest, lb log.Broadcast) {
+func (l *FunctionsListener) handleOracleRequestV1(request *evmrelayTypes.OracleRequest) {
 	defer l.shutdownWaitGroup.Done()
-	l.logger.Infow("oracle request v1 received", "requestID", formatRequestId(request.RequestId))
+	l.logger.Infow("handleOracleRequestV1: oracle request v1 received", "requestID", formatRequestId(request.RequestId))
 	ctx, cancel := l.getNewHandlerContext()
 	defer cancel()
 
 	callbackGasLimit := uint32(request.CallbackGasLimit)
 	newReq := &Request{
 		RequestID:                  request.RequestId,
-		RequestTxHash:              &request.Raw.TxHash,
+		RequestTxHash:              &request.TxHash,
 		ReceivedAt:                 time.Now(),
 		Flags:                      request.Flags[:],
 		CallbackGasLimit:           &callbackGasLimit,
-		CoordinatorContractAddress: &request.Raw.Address,
+		CoordinatorContractAddress: &request.CoordinatorContract,
 	}
 	if err := l.pluginORM.CreateRequest(newReq, pg.WithParentCtx(ctx)); err != nil {
 		if errors.Is(err, ErrDuplicateRequestID) {
-			l.logger.Warnw("received a log with duplicate request ID", "requestID", formatRequestId(request.RequestId), "err", err)
-			l.markLogConsumed(lb, pg.WithParentCtx(ctx))
+			l.logger.Warnw("handleOracleRequestV1: received a log with duplicate request ID", "requestID", formatRequestId(request.RequestId), "err", err)
 		} else {
-			l.logger.Errorw("failed to create a DB entry for new request", "requestID", formatRequestId(request.RequestId), "err", err)
+			l.logger.Errorw("handleOracleRequestV1: failed to create a DB entry for new request", "requestID", formatRequestId(request.RequestId), "err", err)
 		}
 		return
 	}
```

Comment:
IIUC, we no longer need to `markLogConsumed` because we are using the log poller instead of the log broadcaster?

Reply:
yes, LogPollerWrapper takes care of that internally (by keeping track of last block used for queries).